From 674e463d0b270849e27b24453f462464e61f6f7b Mon Sep 17 00:00:00 2001
From: yxj <yxj@rock-chips.com>
Date: Wed, 25 Apr 2012 19:55:42 +0800
Subject: [PATCH] rk30 fb: fix waring initialization from incompatible pointer
 type in board_rk30_sdk.c

	add __ATTR_NULL as the end  of array rkfb_attrs
---
 arch/arm/mach-rk30/board-rk30-sdk.c | 2 +-
 drivers/video/rockchip/rkfb_sysfs.c | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/board-rk30-sdk.c b/arch/arm/mach-rk30/board-rk30-sdk.c
index 03d7bbcddcfa..dcbbc913213a 100644
--- a/arch/arm/mach-rk30/board-rk30-sdk.c
+++ b/arch/arm/mach-rk30/board-rk30-sdk.c
@@ -815,7 +815,7 @@ struct rk29fb_info lcdc1_screen_info = {
 	#if defined(CONFIG_HDMI_RK30)
 	.prop		= EXTEND,	//extend display device
 	.lcd_info  = NULL,
-	.set_screen_info = hdmi_set_info,
+	.set_screen_info = hdmi_init_lcdc,
 	#endif
 };
 #endif
diff --git a/drivers/video/rockchip/rkfb_sysfs.c b/drivers/video/rockchip/rkfb_sysfs.c
index 17f8fafa746a..1b6e74f6ff83 100644
--- a/drivers/video/rockchip/rkfb_sysfs.c
+++ b/drivers/video/rockchip/rkfb_sysfs.c
@@ -159,6 +159,7 @@ static struct device_attribute rkfb_attrs[] = {
 	__ATTR(screen_info, S_IRUGO, show_screen_info, NULL),
 	__ATTR(enable, S_IRUGO | S_IWUSR, show_fb_state, set_fb_state),
 	__ATTR(overlay, S_IRUGO | S_IWUSR, show_overlay, set_overlay),
+	__ATTR_NULL,
 };
 
 int rkfb_create_sysfs(struct fb_info *fbi)
-- 
2.35.3

